pygeoapi_plugins/provider/sensorthings_edr.py

```diff
@@ -0,0 +1,401 @@
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# =================================================================
+
+import logging
+
+from pygeoapi.provider.base import ProviderNoDataError
+from pygeoapi.provider.base_edr import BaseEDRProvider
+from pygeoapi.provider.sensorthings import SensorThingsProvider
+
+LOGGER = logging.getLogger(__name__)
+
+GEOGRAPHIC_CRS = {
+    'coordinates': ['x', 'y'],
+    'system': {
+        'type': 'GeographicCRS',
+        'id': 'http://www.opengis.net/def/crs/OGC/1.3/CRS84',  # noqa
+    },
+}
+
+TEMPORAL_RS = {
+    'coordinates': ['t'],
+    'system': {'type': 'TemporalRS', 'calendar': 'Gregorian'},
+}
+
+
+class SensorThingsEDRProvider(BaseEDRProvider, SensorThingsProvider):
+    def __init__(self, provider_def):
+        """
+        Initialize the SensorThingsEDRProvider instance.
+
+        :param provider_def: provider definition
+        """
+        provider_def['entity'] = 'ObservedProperties'
+        BaseEDRProvider.__init__(self, provider_def)
+        SensorThingsProvider.__init__(self, provider_def)
+        self.expand['ObservedProperties'] = (
+            'Datastreams/Thing/Locations,Datastreams/Observations'  # noqa
+        )
+        self.time_field = 'Datastreams/Observations/resultTime'
+        self._fields = {}
+        self.get_fields()
+
+    def get_fields(self):
+        """
+        Retrieve and store ObservedProperties field definitions.
+
+        :returns: A dictionary mapping field IDs to their properties.
+        """
+        if not self._fields:
+            r = self._get_response(
+                self._url, entity='ObservedProperties', expand='Datastreams'
+            )
+            try:
+                _ = r['value'][0]
+            except IndexError:
+                LOGGER.warning('could not get fields; returning empty set')
+                return {}
+
+            for feature in r['value']:
+                id = str(feature['@iot.id'])
+                key = feature['name']
+                try:
+                    UoM = feature['Datastreams'][0]['unitOfMeasurement']
+                except IndexError:
+                    continue
+
+                self._fields[id] = {
+                    'type': 'number',
+                    'title': key,
+                    'x-ogc-unit': UoM['symbol'],
+                }
+
+        return self._fields
+
+    @BaseEDRProvider.register()
+    def items(self, **kwargs):
+        """
+        Retrieve a collection of items.
+
+        :param kwargs: Additional parameters for the request.
+        :returns: A GeoJSON representation of the items.
+        """
+        pass
+
```

Comment:
Might be worth calling this out since it is not obvious for other contributors.
```suggestion
        """
        # This method is empty due to the way pygeoapi expects EDR providers. 
        # We define this method and then implement `get` inside of the OAF provider
        pass

```